platform-modules/batch-models/src/main/scala/org/ekstep/analytics/job/AssessmentMetricsJob.scala

```diff
@@ -0,0 +1,307 @@
+    val indexName = AssessmentReportUtil.suffixDate(indexPrefix)
+    val indexToEs = AppConf.getConfig("course.es.index.enabled")
+    if (StringUtils.isNotBlank(indexToEs) && StringUtils.equalsIgnoreCase("true", indexToEs)) {
+      AssessmentReportUtil.saveToElastic(indexName, aliasName, reportDF)
+    } else {
+      JobLogger.log("Skipping Indexing assessment report into ES", None, INFO)
+    }
+
+    val result = reportDF
+      .groupBy("courseid")
+      .agg(collect_list("batchid").as("batchid"))
+    val uploadToAzure = AppConf.getConfig("course.upload.reports.enabled")
+    if (StringUtils.isNotBlank(uploadToAzure) && StringUtils.equalsIgnoreCase("true", uploadToAzure)) {
+      val courseBatchList = result.collect.map(r => Map(result.columns.zip(r.toSeq): _*))
+      courseBatchList.foreach(item => {
+        val batchList = item.getOrElse("batchid", null).asInstanceOf[Seq[String]].distinct
+        val courseId = item.getOrElse("courseid", null).toString
+        batchList.foreach(batchId => {
+          if (courseId != null && batchId != null) {
+            val reportData = transposeDF(reportDF, courseId, batchId)
+            // Save report to azure cloud storage
+            AssessmentReportUtil.save(reportData, url)
+          }
+        })
+      })
+    } else {
+      JobLogger.log("Skipping uploading reports into to azure", None, INFO)
+    }
+  }
+
+  /**
+    * Converting rows into  column (Reshaping the dataframe.)
+    * This method converts the name column into header row formate
+    * Example:
+    * Input DF
+    * +------------------+-------+--------------------+-------+-----------+
+    * |              name| userid|            courseid|batchid|total_score|
+    * +------------------+-------+--------------------+-------+-----------+
+    * |Playingwithnumbers|user021|do_21231014887798...|   1001|         10|
+    * |     Whole Numbers|user021|do_21231014887798...|   1001|          4|
+    * +------------------+---------------+-------+--------------------+----
+    *
+    * Output DF: After re-shaping the data frame.
+    * +--------------------+-------+-------+------------------+-------------+
+    * |            courseid|batchid| userid|Playingwithnumbers|Whole Numbers|
+    * +--------------------+-------+-------+------------------+-------------+
+    * |do_21231014887798...|   1001|user021|                10|            4|
+    * +--------------------+-------+-------+------------------+-------------+
+    * Example:
+    */
+  def transposeDF(reportDF: DataFrame, courseId: String, batchId: String): DataFrame = {
+    // Re-shape the dataframe (Convert the content name from the row to column)
+    JobLogger.log(s"Generating report for ${courseId} course and ${batchId} batch")
+    val reshapedDF = reportDF.filter(col("courseid") === courseId && col("batchid") === batchId).
+      groupBy("courseid", "batchid", "userid").pivot("name").agg(first("total_score"))
```

Comment:
1. change "name" to "contentName" 
2. groupBy logic on attemptId instead of "courseid", "bactchid", "userid"